src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -303,8 +308,13 @@ private void OutputFlags(NodeFactory factory, ref ObjectDataBuilder objData)
         {
             UInt16 flags = EETypeBuilderHelpers.ComputeFlags(_type);
 
+            if (_type.GetTypeDefinition() == factory.ArrayOfTEnumeratorType)
+            {
+                // Generic array enumerators use special variance rules recognized by the runtime
+                flags |= (UInt16)EETypeFlags.GenericVarianceFlag;
```

Comment:
The only way I see to get rid of this is to make variance algorithm based. Might be an overkill since we only need it for one very special type.